### PR TITLE
[MINI-5487] Remove old demo app code

### PR DESCRIPTION
**Pull request by laubengaier**
# Description
Removing / cleaning old demo app code 

## Links
https://jira.rakuten-it.com/jira/browse/MINI-5487

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

**Review by Climbatize — APPROVED** [review]

File: Example/AppDelegate.swift [hunk 1]
```diff
@@ -58,14 +40,12 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
     }
 
-    func deepLinkToMiniApp(using token: String) {
-        let rootController = window?.rootViewController as? UINavigationController
-        guard let controllersStack = rootController?.viewControllers else { return }
-        if let homeViewController = controllersStack.first(where: { $0 is ViewController }) as? ViewController {
-            homeViewController.getMiniAppPreviewInfo(previewToken: token, config: Config.current(pinningEnabled: true))
-        } else if let firstLaunchController = controllersStack.first(where: { $0 is FirstLaunchViewController }) as? FirstLaunchViewController {
-            firstLaunchController.previewUsingQRToken = token
-        }
```
Comment (rleojoseph) [inline Example/AppDelegate.swift:68]: Hopefully we are handling the deeplink

File: Example/Analytics/DemoAppAnalytics.swift [hunk 2]
```diff
@@ -74,16 +74,17 @@ public class DemoAppAnalytics {
...
+//            guard let customPermissionCell = cell as? CustomPermissionCell else {
+//                guard let customPermissionCell = cell as? FirstLaunchCustomPermissionCell else {
+//                    return cell.textLabel?.text
+//                }
+//                return customPermissionCell.permissionTitle.text
+//            }
+//            return customPermissionCell.titleLabel?.text
+//        }
+//        return miniAppCell.titleLabel?.text
+        return nil
```
Comment (rleojoseph) [inline Example/Analytics/DemoAppAnalytics.swift:87]: Do we have Demo app analytics maintained in new changes ?

File: Example/Info.plist [hunk 3]
```diff
@@ -4,20 +4,12 @@
...
-	<string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
-	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
-	<key>CFBundleName</key>
-	<string>$(PRODUCT_NAME)</string>
-	<key>CFBundlePackageType</key>
-	<string>APPL</string>
+	<string>com.rakuten.tech.mobile.miniapp.MiniAppDemo</string>
 	<key>CFBundleShortVersionString</key>
-	<string>$(MARKETING_VERSION)</string>
```
Comment (rleojoseph) [inline Example/Info.plist:20]: We need this to differentiate the DEV/STG and prod builds
MARKETING_VERSION = $(RMA_SDK_VERSION)$(RMA_DEMO_APP_BUILD_TYPE)

File: Example/Info.plist [hunk 4]
```diff
@@ -4,20 +4,12 @@
...
 	<key>CFBundleExecutable</key>
 	<string>$(EXECUTABLE_NAME)</string>
 	<key>CFBundleIdentifier</key>
-	<string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
-	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
-	<key>CFBundleName</key>
-	<string>$(PRODUCT_NAME)</string>
-	<key>CFBundlePackageType</key>
-	<string>APPL</string>
```
Comment (rleojoseph) [inline Example/Info.plist:18]: We may need these info too

File: Example/Info.plist [hunk 5]
```diff
@@ -37,27 +29,17 @@
...
 		<true/>
 	</dict>
-	<key>NSCameraUsageDescription</key>
-	<string>This will let you to take Photos</string>
-	<key>NSLocationWhenInUseUsageDescription</key>
-	<string>Demo app helps you find more information near you</string>
-	<key>NSMicrophoneUsageDescription</key>
-	<string>Needs Microphone while recording video</string>
-	<key>NSPhotoLibraryUsageDescription</key>
-	<string>Demo app needs permission to save photos</string>
```
Comment (rleojoseph) [inline Example/Info.plist:54]: Any reason we removed this permissions? We need them for the MiniApps right ?

File: Example/Info.plist [hunk 6]
```diff
@@ -90,20 +72,29 @@
...
 	<string>LaunchScreen</string>
-	<key>UIMainStoryboardFile</key>
-	<string>Main</string>
-	<key>UIRequiredDeviceCapabilities</key>
-	<array>
-		<string>armv7</string>
-	</array>
 	<key>UISupportedInterfaceOrientations</key>
 	<array>
-		<string>UIInterfaceOrientationLandscapeRight</string>
```
Comment (rleojoseph) [inline Example/Info.plist:103]: We need to support other orientations as well, when playing a full screen video in MiniApp, they support all orientations.

**Review by rleojoseph — COMMENTED** [review]
Overall changes looks fine. I have few concerns for which I mentioned in the comments. Please have a look at it.
Also, please mention in the Changelog that we removed old code Demo app

File: Example/AppDelegate.swift (hunk 1 above)
Comment (laubengaier) [reply Example/AppDelegate.swift:68]: yeah I've moved it here https://github.com/laubengaier/ios-miniapp/blob/feat-5.0.0/remove-old-demo-app-code/Example/SampleApp.swift#L18

File: Example/Info.plist (hunk 3 above)
Comment (laubengaier) [reply Example/Info.plist:20]: @rleojoseph regarding plist it got an update and is now automatically removing these from `Info.plist` and putting them into the project file (https://github.com/rakutentech/ios-miniapp/blob/26c4cc05a4fd00788d9da247e386f9de833b3faf/MiniAppSPM/Sample%20SPM.xcodeproj/project.pbxproj#L1115) although some keys are missing I'll check

File: Example/Analytics/DemoAppAnalytics.swift (hunk 2 above)
Comment (laubengaier) [reply Example/Analytics/DemoAppAnalytics.swift:87]: as discussed I'll address it in a new PR

**Review by rleojoseph — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**